objects/src/notes/mod.rs

```diff
@@ -200,33 +204,34 @@ impl From<&Note> for Vec<Felt> {
     ///     out[-9..-5]   = origin.SUB_HASH
     ///     out[-5..-1]   = origin.NOTE_ROOT
     ///     out[-1]       = origin.node_index
-    fn from(note: &Note) -> Self {
-        // compute capacity of the output vector.  If we have an odd number of assets, we need to
-        // pad the output with an empty word.
-        let capacity = if note.vault.num_assets() % 2 == 1 {
-            30 + (note.vault.num_assets() + 1) * 4
-        } else {
-            30 + note.vault.num_assets() * 4
-        };
-
-        let mut out = Vec::with_capacity(capacity);
-
-        out.extend_from_slice(&note.serial_num);
-        out.extend_from_slice(note.script.hash().as_elements());
-        out.extend_from_slice(note.inputs.hash().as_elements());
-        out.extend_from_slice(note.vault.hash().as_elements());
-        out.extend_from_slice(&Word::from(note.metadata()));
-        out.extend(note.vault.to_padded_assets());
-
-        // TODO: this is a temporary solution - replace with TryFrom
-        let origin = note.origin().as_ref().unwrap();
-
-        // populate origin data
-        out.push(origin.block_num());
-        out.extend(Word::from(origin.sub_hash()));
-        out.extend(Word::from(origin.note_root()));
-        out.push(Felt::from(origin.node_index().value()));
-
-        out
+    fn to_advice_inputs<T: AdviceInputsBuilder>(&self, target: &mut T) {
+        // push core data onto the stack
+        target.push_onto_stack(&self.serial_num);
+        target.push_onto_stack(self.script.hash().as_elements());
+        target.push_onto_stack(self.inputs.hash().as_elements());
+        target.push_onto_stack(self.vault.hash().as_elements());
+        target.push_onto_stack(&Word::from(self.metadata()));
+
+        // add assets to the stack and advice map
+        target.push_onto_stack(&self.vault.to_padded_assets());
+        target.insert_into_map(self.vault.hash().into(), self.vault.to_padded_assets());
+
+        // origin must be populated for created notes
+        let origin = self.origin().as_ref().unwrap();
```

Comment:
> nit: should we use an `expect()` here?

Yes good point `expect()` would be better here, I will update it.

> Also, a thought for the future: maybe `to_advice_inputs()` should return a `Result` to handle situations like this (though, not sure what the error type would be).

Yes we could consider modifying to return `Result`.  However, in this case I feel like the procedure should be infallible.  Maybe a cleaner approach would be to remove `NoteOrigin` from the `Note` object and implement a new parent struct:

```rust
pub struct NoteWithOrigin {
    note: Note,
    origin: NoteOrigin
}
```

We can then implement `ToAdviceInputs` on this struct. This would be infallible.  What do you think?